### PR TITLE
feature: adding optional results in kmeans infer

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
# Description
Adding optional results to align predict behavior between scikit-learn-intelex and daal4py

### CODE REVIEW

**Review by inteldimitrius — APPROVED** [review]
LGTM well done. Please make sure that CI is green

File: cpp/oneapi/dal/algo/kmeans/backend/gpu/infer_kernel_dpc.cpp [hunk 1]
```diff
@@ -46,7 +46,9 @@ struct infer_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
         auto arr_centroids = pr::table2ndarray<Float>(queue,
                                                       input.get_model().get_centroids(),
                                                       sycl::usm::alloc::device);
-        auto result = infer_result<task::clustering>{};
+        auto result =
+            infer_result<task::clustering>{}.set_result_options(desc.get_result_options());
+        ;
```
Comment (Vika-F) [inline cpp/oneapi/dal/algo/kmeans/backend/gpu/infer_kernel_dpc.cpp:51]:
```suggestion
```
excessive semicolon

File: cpp/oneapi/dal/algo/kmeans/common.hpp [hunk 2]
```diff
@@ -52,6 +53,34 @@ using v1::by_default;
...
+ONEDAL_EXPORT result_option_id get_compute_assignments_id();
+ONEDAL_EXPORT result_option_id get_compute_exact_objective_function_id();
+
+} // namespace detail
+
+/// Result options are used to define
+/// what should an algorithm return
+namespace result_options {
+
+/// Return assignments
```
Comment (Vika-F) [inline cpp/oneapi/dal/algo/kmeans/common.hpp:76]:
```suggestion
/// Return the assignments
```

File: cpp/oneapi/dal/algo/kmeans/common.hpp [hunk 3]
```diff
@@ -52,6 +53,34 @@ using v1::by_default;
...
+const inline result_option_id compute_assignments = detail::get_compute_assignments_id();
+/// Return objective function
```
Comment (Vika-F) [inline cpp/oneapi/dal/algo/kmeans/common.hpp:78]:
```suggestion
/// Return the objective function
```

**Review by Vika-F — APPROVED** [review]
The code looks good to me. Please make the minor changes requested and LGTM.

**Pull request merged**